src/main/java/core/basesyntax/JavaDateTimeApi.java

```diff
@@ -1,125 +1,177 @@
 package core.basesyntax;
 
+import java.time.DateTimeException;
+import java.time.Instant;
 import java.time.LocalDate;
 import java.time.LocalDateTime;
 import java.time.LocalTime;
 import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
 import java.util.Optional;
 
 public class JavaDateTimeApi {
+    public static final int FEBRUARY_DAYS = 28;
+    private static final int NUMBER_OF_DAYS = 31;
+    private static final int NUMBER_OF_MONTHS = 12;
+    private static final int ARRAY_YEAR = 0;
+    private static final int ARRAY_MONTH = 1;
+    private static final int ARRAY_DAY = 2;
+    private static final LocalDate NOW = LocalDate.now();
+    private static final String DATE_TIME_FORMAT = "dd MMMM yyyy HH:mm";
+    private static final String UKRAINE_TIMEZONE = "+02:00";
+    private static final int FEBRUARY = 2;
+
     /**
      * Return the current date as a String depending on a query.
-     *
+     * <p>
      * The query can be passed for the whole date or for it's part:
-     *      - FULL - current date as a whole: year, month, day of month
-     *        formatted as `YYYY-MM-DD` (a default return value);
-     *      - YEAR - current year;
-     *      - MONTH - name of the current month;
-     *      - DAY - current day of month;
+     * - FULL - current date as a whole: year, month, day of month
+     * formatted as `YYYY-MM-DD` (a default return value);
+     * - YEAR - current year;
+     * - MONTH - name of the current month;
+     * - DAY - current day of month;
      * In any other case throw DateTimeException.
      **/
     public String todayDate(DateTimePart datePart) {
-        return "Today";
+        switch (datePart) {
+            case FULL:
+                return String.valueOf(NOW);
+            case YEAR:
+                return String.valueOf(NOW.getYear());
+            case MONTH:
+                return String.valueOf(NOW.getMonth());
+            case DAY:
+                return String.valueOf(NOW.getDayOfMonth());
+            default:
+                throw new DateTimeException("No such date part");
+        }
     }
 
     /**
      * Given an Array of 3 elements, where
-     *         - 1-st element is a `year`;
-     *         - 2-nd element is s `month`;
-     *         - 3-rd element is a `day of month`;
-     *
+     * - 1-st element is a `year`;
+     * - 2-nd element is s `month`;
+     * - 3-rd element is a `day of month`;
+     * <p>
      * Return Optional of a date built from these elements.
      */
     public Optional<LocalDate> getDate(Integer[] dateParams) {
-        return Optional.empty();
+        if (dateParams.length == 0
+                || dateParams[ARRAY_DAY] > NUMBER_OF_DAYS
+                || dateParams[ARRAY_MONTH] > NUMBER_OF_MONTHS
+                || (dateParams[ARRAY_MONTH] == FEBRUARY && dateParams[ARRAY_DAY] > FEBRUARY_DAYS)) {
```

Comment:
no need to make these checks, try catch will do it

Reply:
I got it now) Read about this exception and became clear to me finally) Done